Rich_Text_Example.c

```diff
@@ -0,0 +1,270 @@
+ * 
+ * Apply font/text atrributes on text (bold, italic, color, font-size)
+*/
+
+#define EFL_BETA_API_SUPPORT 1
+ 
+#include <Eina.h>
+#include <Elementary.h>
+#include <Efl_Ui.h>
+
+#define COMP_TEXT(obj, text)  (strcmp(efl_text_get(efl_content_get((Eo*)obj)), text) == 0)
+
+#define CHECK_BOX_BOLD "Bold"
+#define CHECK_BOX_ITALIC "Italic"
+#define CHECK_BOX_RED "Red"
+#define CHECK_BOX_30 "30"
+
+/************ Rich Text Logic ******************/
+
+static Eo *attribute_factory = NULL;
+static Eo *check_bold, *check_italic, *check_red, *check_30;
+static Eo *cursor_main, *cursor_start, *cursor_end, *sel_start, *sel_end;
+
+static void
+_apply_attribute(Eo *ui_text)
+{
+   /*
+     We have three cases :
+      1- If there is Selection, we insert attribute for the selected area.
+      2- If the main cursor is inside a word, we insert the attribute for the whole word
+      3- If outside word, newly inserted text at current posistion should have the chosen attributes 
+    */
+   
+   if (efl2_text_cursor_equal(sel_start, sel_end)) // if no selection
+     {
+         efl2_text_cursor_copy(cursor_main, cursor_start);
+         efl2_text_cursor_copy(cursor_main, cursor_end);
+
+         efl2_text_cursor_word_start(cursor_start);
+         efl2_text_cursor_word_end(cursor_end);
+
+         int start_pos, end_pos, main_pos;
+         
+         start_pos = efl2_text_cursor_position_get(cursor_start);
+         end_pos = efl2_text_cursor_position_get(cursor_end);
+         main_pos = efl2_text_cursor_position_get(cursor_main);
+
+         if ( main_pos > start_pos && main_pos < end_pos) // inside word
+           {
+              efl2_text_attribute_factory_insert(attribute_factory, cursor_start, cursor_end);
+           }
+         else
+           {
+              //if before/after word newly inserted text by user should use chosen attributes, how to do it?
+           }
+     }
+     else
+       efl2_text_attribute_factory_insert(attribute_factory, sel_start, sel_end);
+}
+
+static void
+_update_check_boxes()
+{
+   efl_ui_selectable_selected_set(check_bold, efl2_text_font_properties_weight_get(attribute_factory) == EFL2_TEXT_FONT_WEIGHT_BOLD);
+   efl_ui_selectable_selected_set(check_italic, efl2_text_font_properties_font_slant_get(attribute_factory) == EFL2_TEXT_FONT_SLANT_ITALIC);
+   efl_ui_selectable_selected_set(check_30, efl2_text_font_properties_font_size_get(attribute_factory) == 30);
+
+   unsigned char r,g,b,a;
+   efl2_text_style_foreground_color_get(attribute_factory, &r, &g, &b, &a);
+   if (r == 255 && g == 0 && b == 0 && a == 255)
+     efl_ui_selectable_selected_set(check_red, EINA_TRUE);
+   else
+     efl_ui_selectable_selected_set(check_red, EINA_FALSE);
+}
+
+static void
+_attribute_text_get(Eo *cur_start, Eo *cur_end)
+{
+   Efl2_Text_Attribute_Handle *handle;
+   Eina_Iterator *itr;
+
+   itr = efl2_text_attribute_factory_range_attributes_get(cur_start, cur_end); // Does it work if cur_start == cur_end ?

```

Comment:
Does it work if cur_start == cur_end ?
I want to get attributes on cursor position